### DISCUSSION TITLE
Problems with translation of PS 1.7, 8.0

### DISCUSSION THREAD

**Discussion by hujax** [PrestaShop/PrestaShop#32815] [Q&A]
Hello! I would like to understand what is going on with Prestashop. I have been working with this engine for 6 years and everything was fine until version 1.7.6, when the translation system became more complicated. At first it looked like a bug, but now it's obvious that the developers did this intentionally. Why?

PS is a free engine, right? You can download and use it today. It's a very powerful product, but the basic package lacks many options. If you install new themes and modules that solve these problems, how do you translate them? I'm not the first or tenth person to write about the monstrous difficulties when trying to translate the right module or theme. Instead of the simple and straightforward logic that was in 1.6, every new version has a problem with translations. The Translations menu simply doesn't have the right lines. And people get creative, fix the code, come up with workarounds - but this problem didn't exist before, why create it?

It is logical and understandable that developers want to make money on themes and modules - but why after the purchase the user must go through seven circles of hell to configure them the way he likes? Or is the whole point of this innovation to make the user either learn how to program or pay to customize the translation as well? So write about it directly! I have a lot of respect for Prestashop, but this downgrade blows my mind.

**Comment by matks** [comment]
Hello @hujax what can I tell you ..?

I think you obviously know that, no, there is no evil plan to downgrade the software to make people buy stuff, no, there is no evil plan to bring PrestaShop to ruin or whatever.

You seem to like the translation system that was in PS 1.6 but it was faulty too. However either you did not encounter the problems on PS 1.6 or you learnt how to avoid / workaround them.

If you want to help, don't hesitate to contribute. You can help fixing bugs, you can help designing a better translation system, you could even introduce smart interfaces in the software that would allow using either 1.6 system or newer system. If you like PrestaShop, then do something about it.

But ranting about the people who are working on it is going to do nothing.

If you find that any translation problems are critical for you, then you can _invest_ into having it done. You can hire a developer to fix it, or if you are a developer yourself, you can try doing it on your own. The most important thing is to share that fix with everyone by submitting a Pull Request -- that's what the open source spirit is all about.

> It is logical and understandable that developers want to make money on themes and modules - but why after the purchase the user must go through seven circles of hell to configure them the way he likes?

For example you could start working on that.
